core/src/test/java/org/apache/iceberg/puffin/TestPuffinFormat.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.puffin;
+
+import java.io.ByteArrayOutputStream;
+import org.junit.Test;
+
+import static org.apache.iceberg.puffin.PuffinFormat.readIntegerLittleEndian;
+import static org.apache.iceberg.puffin.PuffinFormat.writeIntegerLittleEndian;
+import static org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestPuffinFormat {
```

Comment:
Rather than hard-coding, or at least in addition, I think these tests should validate against the behavior of reading the little-endian value from a ByteBuffer. That will help readers see that it is correct.

Reply:
i am not sure i understand, please exemplify

Reply:
I think you should also validate these tests using the standard library. For example:

```java
  byte[] leBytes = bytes(42,0,0,0);
  ByteBuffer.wrap(leBytes).order(ByteOrder.LITTLE_ENDIAN).getInt() == readIntegerLittleEndian(leBytes, 0)
```